codeflash/code_utils/version_check.py

```diff
@@ -0,0 +1,96 @@
+"""Version checking utilities for codeflash."""
+
+import sys
+import time
+from typing import Optional
+
+import requests
+from packaging import version
+
+from codeflash.cli_cmds.console import console, logger
+from codeflash.version import __version__
+
+# Simple cache to avoid checking too frequently
+_version_cache = {"version": None, "timestamp": 0}
+_cache_duration = 3600  # 1 hour cache
+
+
+def get_latest_version_from_pypi() -> Optional[str]:
+    """Get the latest version of codeflash from PyPI.
+    
+    Returns:
+        The latest version string from PyPI, or None if the request fails.
+    """
+    global _version_cache
+    
+    # Check cache first
+    current_time = time.time()
+    if (_version_cache["version"] is not None and 
+        current_time - _version_cache["timestamp"] < _cache_duration):
+        return _version_cache["version"]
+    
+    try:
+        response = requests.get("https://pypi.org/pypi/codeflash/json", timeout=10)
```

Comment:
nit: can have quick timeout here to make it non blocking if there is network issue.